### PR TITLE
fix: refactor OG link builder

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Fixes #XXXX (GitHub issue number)
- Fixes CAL-XXXX (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Mandatory Tasks (DO NOT REMOVE)

- [ ] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Checklist

- I haven't read the [contributing guide](https://github.com/calcom/cal.com/blob/main/CONTRIBUTING.md)
- My code doesn't follow the style guidelines of this project
- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if my changes generate no new warnings

### CODE REVIEW

File: packages/lib/OgImages.tsx [hunk 1]
```diff
@@ -58,12 +58,12 @@ const makeAbsoluteUrl = (url: string) => (/^https?:\/\//.test(url) ? url : `${CA
  */
 export const constructMeetingImage = ({ title, users = [], profile }: MeetingImageProps): string => {
   const url = [
```
Comment (keithwillcode) [inline packages/lib/OgImages.tsx:60]: Can we not encode the entire URL parameter instead of each individual component?

Comment (hbjORbj) [reply packages/lib/OgImages.tsx]: addressed

**Review by keithwillcode — DISMISSED** [review]
Code changes look ok. I won’t personally have time to test today so let’s make sure everything looks good before shipping.

**Comment by hbjORbj** [comment]
Blocked by the failing integration test. Seems like there's a problem with the test.

**Review by emrysal — APPROVED** [review]
Hard to confirm it works, but it'll work better than it currently does 😆 - changes look good though.

**Pull request merged**